src/NUnitFramework/tests/TestContextTests.cs

```diff
@@ -459,35 +460,31 @@ public void TearDown()
     public class TestContextOneTimeTearDownTests
     {
         [Test]
-        public void TestTruth()
+        public void TestContextContainsSuiteResults()
         {
-            Assert.That(true, Is.True);
-        }
+            var fixture = new TestContextOneTimeTearDownData();
 
-        [Test]
-        public void TestFalsehood()
-        {
-            Assert.That(false, Is.False);
-        }
+            var result = TestBuilder.RunTestFixture(fixture);
 
-        [Test, Explicit]
-        public void TestExplicit()
-        {
-            Assert.Pass("Always passes if you run it!");
-        }
+            Assert.That(result, Is.Not.Null);
+            Assert.Multiple(() =>
+            {
+                Assert.That(result.ResultState, Is.EqualTo(ResultState.Success));
+                Assert.That(result.PassCount, Is.EqualTo(2));
+                Assert.That(result.FailCount, Is.EqualTo(0));
+                Assert.That(result.SkipCount, Is.EqualTo(1));
+            });
 
-        [OneTimeTearDown]
-        public void OneTimeTearDown()
-        {
-            TestContext context = TestContext.CurrentContext;
-            Assert.That(context, Is.Not.Null);
-            Assert.That(context.Test, Is.Not.Null);
-            Assert.That(context.Test.Name, Is.EqualTo("TestContextOneTimeTearDownTests"));
-            Assert.That(context.Result, Is.Not.Null);
-            Assert.That(context.Result.Outcome, Is.EqualTo(ResultState.Success));
-            Assert.That(context.Result.PassCount, Is.EqualTo(2));
-            Assert.That(context.Result.FailCount, Is.EqualTo(0));
-            Assert.That(context.Result.SkipCount, Is.EqualTo(1));
+            var tearDownResult = fixture.ResultInOneTimeTearDown;
+
+            Assert.That(tearDownResult, Is.Not.Null);
+            Assert.Multiple(() =>
+            {
+                Assert.That(tearDownResult!.Outcome, Is.EqualTo(result.ResultState));
```

Comment:
You shouldn't need the null suppression operator `!`.

Reply:
I can try to take a closer look at this one tomorrow. I had added it to suppress a warning, but I can also try updating the referenced version of the analyzer package. I think I saw that we currently reference 4.3.0 whereas latest is 4.6.0

Reply:
Aha. Appears I was still a few bugfix versions behind on Visual Studio. I've upgraded to latest and now no longer need the `!`. Thanks for pointing that out!

I've pushed the change to remove it and will merge once the builds pass